lib/presentation/ui/home/home_bloc.dart

```diff
@@ -72,17 +73,47 @@ class HomeBloc extends BaseBloc with RefreshBlocMixin {
   var _occurrencesListSubject =
       BehaviorSubject<List<OccurrenceModel>>.seeded([]);
 
+  /// Subject that will fetch additional pages
+  var _fetchNextPageSubject = PublishSubject<Event>();
+  Sink<Event> get fetchNextPageSink => _fetchNextPageSubject.sink;
+
+  /// Subject that keeps track of the page number
+  var _pageNumberSubject = BehaviorSubject<int>.seeded(1);
+
   HomeBloc(this._occurrenceManager, this._sharedPreferencesManager) {
     disposable.add(_fetchNewDataSubject.stream
+        .doOnData((_) => showLoading())
+        .doOnData((_) => _pageNumberSubject.add(1))
         .flatMap((_) => _occurrenceManager.getRecentOccurrences())
```

Comment:
Why do you need a `flapMap` here instead of just `map`?

Reply:
The `flatMap` is needed because we are returning an `Observable`.

`map` is used to map data from `T` to `R`, `flatMap` is used to chain `Observable<T>` to `Observable<R>` so that they can be called in succession.